pkg/parse/root_test.go

```diff
@@ -1788,3 +1788,59 @@ func TestSummarizeErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestPrependRootSyncRemediatorStatus(t *testing.T) {
+	const rootSyncName = "my-root-sync"
+	const thisManager = "this-manager"
+	const otherManager = "other-manager"
+	conflictingObject := fake.NamespaceObject("foo-ns",
+		core.Annotation(metadata.ResourceManagerKey, otherManager))
+	conflictError := status.ManagementConflictErrorWrap(conflictingObject, thisManager)
+	testCases := map[string]struct {
+		thisSyncErrors []v1beta1.ConfigSyncError
+		expectedErrors []v1beta1.ConfigSyncError
+	}{
+		"empty errors": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+		},
+		"unchanged conflict error": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.ToCSE(),
+			},
+		},
+		"inverted conflict error": {
+			thisSyncErrors: []v1beta1.ConfigSyncError{
+				conflictError.CurrentManagerError().ToCSE(),
+			},
+			expectedErrors: []v1beta1.ConfigSyncError{
+				conflictError.CurrentManagerError().ToCSE(),
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			rootSync := fake.RootSyncObjectV1Beta1(rootSyncName)
+			rootSync.Status.Sync.Errors = tc.thisSyncErrors
+			fakeClient := syncertest.NewClient(t, core.Scheme, rootSync)
+			ctx := context.Background()
+			err := prependRootSyncRemediatorStatus(ctx, fakeClient, rootSyncName,
+				[]status.ManagementConflictError{conflictError}, defaultDenominator)
+			if err != nil {
+				t.Fatal(err)
+			}
+			var updatedRootSync v1beta1.RootSync
+			if err := fakeClient.Get(ctx, rootsync.ObjectKey(rootSyncName), &updatedRootSync); err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(tc.expectedErrors, updatedRootSync.Status.Sync.Errors); diff != "" {
```

Comment:
nit: cmp.Diff works, but I like the output from testutil.AssertEqual better. It shows the left and right and then diff. plus it's shorter to write.

Reply:
done